application/models/Survey.php

```diff
@@ -297,6 +297,12 @@ public function delete($recursive=true)
             // Delete all uploaded files.
             rmdirr(Yii::app()->getConfig('uploaddir').'/surveys/'.$this->sid);
         }
+
+        // Remove from cache
+        if (array_key_exists($this->sid, $this->findByPkCache)) {
+            unset $this->findByPkCache[$this->sid];
```

Comment:
Unset need `()`?

Reply:
My bad! Parenthesis missing. Sorry. Seems I overviewed that.

Unset is needed as to remove the survey from the cache if it is deleted.
Although cached will usually be refreshed as page is refreshed after delete, this code will help avoiding future problems, for example when testing.

I think is theorically needed and practically usefull in few situtations. But still, needed I think